packages/core/logging/core-logging-server-internal/src/appenders/rolling_file/retention/retention_policy.ts

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { dirname, join } from 'path';
+import { schema } from '@kbn/config-schema';
+import type { RetentionPolicyConfig } from '@kbn/core-logging-server';
+import type { RollingFileContext } from '../rolling_file_context';
+import { deleteFiles } from './fs';
+import { listFilesExceedingSize, listFilesOlderThan } from './utils';
+
+export const retentionPolicyConfigSchema = schema.object(
+  {
+    maxFiles: schema.maybe(schema.number({ min: 1, max: 365 })),
+    maxAccumulatedFileSize: schema.maybe(schema.byteSize()),
+    removeOlderThan: schema.maybe(schema.duration({ max: '365d' })),
+  },
+  {
+    validate: (config) => {
+      if (!config.maxFiles && !config.maxAccumulatedFileSize && !config.removeOlderThan) {
+        return 'Retention policy must define at least one directive: maxFiles, maxAccumulatedFileSize or removeOlderThan';
+      }
```

Comment:
I'm allowing multiple directives to be defined at the same time (and the implementation follows), because I didn't see any reason not to.